lib/libc/Kconfig

```diff
@@ -11,6 +11,14 @@ config REQUIRES_FULL_LIBC
 	  Helper symbol to indicate some feature requires a C library implementation
 	  with more functionality than what MINIMAL_LIBC provides
 
+config SUPPORT_FULL_LIBC
```

Comment:
I think we have some inconsistencies in the naming of the helper symbols indicating the support status of LIBCs."

After the changes in this PR, we have:

* SUPPORT_MINIMAL_LIBC
* SUPPORT_FULL_LIBC
* NEWLIB_SUPPORTED
* PICOLIBC_SUPPORTED

I would like to settle on just one naming convention for these symbols, and my personal preference is to use the `*_SUPPORTED`, such that we end up with:

* MINIMAL_LIBC_SUPPORTED
* NEWLIB_LIBC_SUPPORTED
* PICOLIBC_SUPPORTED
* ARCMWDT_LIBC_SUPPORTED
* FULL_LIBC_SUPPORTED

So, every LIBC type will have an associated `_SUPPORTED` symbol that indicates whether the particular LIBC is supported.

If we do this, I think it makes sense to keep the `NEWLIB_LIBC_SUPPORTED` symbol as `default y, depends on TOOLCHAIN_HAS_NEWLIB` for the sake of generalisation -- by doing this, we do not hard-code the support status of `NEWLIB_LIBC` (or any other LIBC for that matter) as simple `TOOLCHAIN_HAS`; it is foreseeable that it may be necessary to add more dependencies (e.g. support status per arch) for some LIBCs in the future.

We can also replicate this pattern to the C++ library side such that we have `GLIBCXX_LIBCPP_SUPPORTED` and so on (just thinking about the bigger picture, not asking this to be done in this PR; I will do that later).

Reply:
That does look like a nice clean up.

Reply:
What about the REQUIRES symbols like `REQUIRES_FULL_LIBC`. By symmetry, should these be `FULL_LIBC_REQUIRED`?